test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeControllerSpec.scala

```diff
@@ -1562,104 +1563,136 @@ class RepresenteeControllerSpec
 
-          "that question hasn't been answered yet" in {
-            inSequence {
-              mockAuthWithNoRetrievals()
-              mockGetSession(
-                sessionWithFillingOutReturn(allQuestionsAnswers.copy(dateOfDeath = None), Left(PersonalRepresentative))._1
-              )
-            }
-
-            checkIsRedirect(performAction(), routes.RepresenteeController.enterDateOfDeath())
+        "that question hasn't been answered yet" in {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              sessionWithFillingOutReturn(allQuestionsAnswers.copy(dateOfDeath = None), Left(PersonalRepresentative))._1
+            )
           }
+
+          checkIsRedirect(performAction(), routes.RepresenteeController.enterDateOfDeath())
         }
+      }
 
-        "redirect to the check contact details page" when {
+      "redirect to the check contact details page" when {
 
-          "there are no contact details in session" in {
-            inSequence {
-              mockAuthWithNoRetrievals()
-              mockGetSession(
-                sessionWithFillingOutReturn(
-                  allQuestionsAnswers.copy(contactDetails = None),
-                  Left(PersonalRepresentative)
-                )._1
-              )
-            }
-
-            checkIsRedirect(
-              performAction(),
-              routes.RepresenteeController.checkContactDetails()
+        "there are no contact details in session" in {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              sessionWithFillingOutReturn(
+                allQuestionsAnswers.copy(contactDetails = None),
+                Left(PersonalRepresentative)
+              )._1
             )
           }
 
-          "the user has not confirmed the contact details" in {
-            inSequence {
-              mockAuthWithNoRetrievals()
-              mockGetSession(
-                sessionWithFillingOutReturn(
-                  allQuestionsAnswers.copy(hasConfirmedContactDetails = false),
-                  Right(Capacitor)
-                )._1
-              )
-            }
+          checkIsRedirect(
+            performAction(),
+            routes.RepresenteeController.checkContactDetails()
+          )
+        }
 
-            checkIsRedirect(
-              performAction(),
-              routes.RepresenteeController.checkContactDetails()
+        "the user has not confirmed the contact details" in {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              sessionWithFillingOutReturn(
+                allQuestionsAnswers.copy(hasConfirmedContactDetails = false),
+                Right(Capacitor)
+              )._1
             )
           }
 
+          checkIsRedirect(
+            performAction(),
+            routes.RepresenteeController.checkContactDetails()
+          )
         }
+
+      }
```

Comment:
we're missing a tests for the transformation of incomplete answers to complete and what happens when the storing the draft return and session fails. And also tests for the content on the cya page itself.